string text
byte[] image
string image_format